subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeSwiftExternalSourceDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,245 @@
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'swift-library'
+            apply plugin: 'xcode'
+            apply plugin: 'xctest'
+
+            dependencies {
+                testImplementation "org.test:greeter:latest.integration"
+            }
+        """
+        test.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppSharedLibrary", ":xcode")
+        rootXcodeWorkspace.contentFile.assertHasProjects("${rootProjectName}.xcodeproj")
+
+        def appProject = xcodeProject("${rootProjectName}.xcodeproj").projectFile
+        appProject.targets.find { it.isUnitTest() }.getBuildSettings().SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file('build/modules/main/debug'), checkoutDir(repo.name, commit.id.name, repo.id).file('build/modules/main/debug'))
+        appProject.indexTarget.getBuildSettings().SWIFT_INCLUDE_PATHS == toSpaceSeparatedList(file('build/modules/main/debug'), checkoutDir(repo.name, commit.id.name, repo.id).file('build/modules/main/debug'))
+    }
+
+    def "adds source dependencies Xcode project of test component to Xcode workspace"() {
+        def library = new SwiftLib()
+        def test = new SwiftLibTest(library, library.greeter, library.sum, library.multiply)
+
+        given:
+        buildTestFixture.withBuildInSubDir()
+        depProject = singleProjectBuild("greeter") {
+            buildFile << """
+                apply plugin: 'swift-library'
+                apply plugin: 'xcode'
+            """
+            library.writeToProject(it)
+        }
+        def commit = repo.commit('initial commit', GFileUtils.listFiles(file('greeter'), null, true))
+
+
+        and:
+        settingsFile << """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:greeter") {
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'swift-library'
+            apply plugin: 'xcode'
+            apply plugin: 'xctest'
+
+            dependencies {
+                testImplementation "org.test:greeter:latest.integration"
+            }
+        """
+        test.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppSharedLibrary", ":xcode",
+            ":greeter:xcodeProject", ":greeter:xcodeProjectWorkspaceSettings", ":greeter:xcodeSchemeGreeterSharedLibrary")
```

Comment:
should build the binaries of the `greeter` library, rather than the project files.